### PR TITLE
TRUNK-6202 Replace Hibernate Criteria API with JPA for HibernateMedicationDispenseDAO

### PR DESCRIPTION
**Pull request by k4pran**
## Description of what I changed

This task is to migrate the deprecated hibernate criteria to jpa criteria for the HibernateMedicationDispenseDAO class.

## Issue I worked on

see https://issues.openmrs.org/browse/TRUNK-

## Checklist: I completed these to help reviewers :)

- [X] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [X] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [X] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [X] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [X] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

File: api/src/main/java/org/openmrs/api/db/hibernate/HibernateMedicationDispenseDAO.java [hunk 1]
```diff
@@ -41,17 +41,14 @@ public MedicationDispense getMedicationDispense(Integer medicationDispenseId) {
...
-			.uniqueResult();
+		return HibernateUtil.getUniqueEntityByUUID(sessionFactory, MedicationDispense.class, uuid);
 	}
 
 	@Override
 	public List<MedicationDispense> getMedicationDispenseByCriteria(MedicationDispenseCriteria criteria) {
 		CriteriaBuilder criteriaBuilder = sessionFactory.getCurrentSession().getCriteriaBuilder();
-		CriteriaQuery<MedicationDispense> criteriaQuery = criteriaBuilder.createQuery(MedicationDispense.class);
-		Root<MedicationDispense> md = criteriaQuery.from(MedicationDispense.class);
+		CriteriaQuery<MedicationDispense> cq = criteriaBuilder.createQuery(MedicationDispense.class);
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/api/db/hibernate/HibernateMedicationDispenseDAO.java:50]: What is the motivation for changing the variable name from `criteriaQuery` to `cq`?

Comment (k4pran) [reply api/src/main/java/org/openmrs/api/db/hibernate/HibernateMedicationDispenseDAO.java:50]: In one of the other PRs you mentioned a few options on the naming conventions for these and we had been using cq in the other PRs

Comment (dkayiwa) [reply api/src/main/java/org/openmrs/api/db/hibernate/HibernateMedicationDispenseDAO.java:50]: Very true indeed! But the challenge here is that the author of this code used `criteriaQuery` which looks like a better variable name than `cq`. So when they notice that we are changing from a clearer name, don't you think that they will feel like we are making their code less readable? 😊

Comment (k4pran) [reply api/src/main/java/org/openmrs/api/db/hibernate/HibernateMedicationDispenseDAO.java:50]: Yea that makes sense, I can roll it back

**Pull request merged**